src/backend/base/langflow/services/tracing/arize_phoenix.py

```diff
@@ -0,0 +1,323 @@
+                )
+
+            if enable_phoenix_tracing:
+                tracer_provider.add_span_processor(
+                    span_processor=span_processor(
+                        span_exporter=HTTPSpanExporter(
+                            endpoint=phoenix_endpoint,
+                            headers=phoenix_headers,
+                        ),
+                    )
+                )
+
+            self.tracer_provider = tracer_provider
+        except ImportError:
+            logger.exception(
+                "Could not import arize-phoenix-otel. Please install it with `pip install arize-phoenix-otel`."
+            )
+            return False
+
+        try:
+            from openinference.instrumentation.langchain import LangChainInstrumentor
+
+            LangChainInstrumentor().instrument(tracer_provider=self.tracer_provider, skip_dep_check=True)
+        except ImportError:
+            logger.exception(
+                "Could not import LangChainInstrumentor."
+                "Please install it with `pip install openinference-instrumentation-langchain`."
+            )
+            return False
+
+        return True
+
+    @override
+    def add_trace(
+        self,
+        trace_id: str,
+        trace_name: str,
+        trace_type: str,
+        inputs: dict[str, Any],
+        metadata: dict[str, Any] | None = None,
+        vertex: Vertex | None = None,
+    ) -> None:
+        """Adds a trace span, attaching inputs and metadata as attributes."""
+        if not self._ready:
+            return
+
+        span_context = self.propagator.extract(carrier=self.carrier)
+        child_span = self.tracer.start_span(
+            name=trace_name,
+            context=span_context,
+            start_time=self._get_current_timestamp(),
+        )
+
+        if trace_type == "prompt":
+            child_span.set_attribute(SpanAttributes.OPENINFERENCE_SPAN_KIND, "chain")
+        else:
+            child_span.set_attribute(SpanAttributes.OPENINFERENCE_SPAN_KIND, trace_type)
+
+        if "session_id" in inputs and len(inputs["session_id"]) > 0 and inputs["session_id"] != self.flow_id:
+            child_span.set_attribute(SpanAttributes.SESSION_ID, inputs["session_id"])
+        else:
+            child_span.set_attribute(SpanAttributes.SESSION_ID, self.flow_id)
+
+        processed_inputs = self._convert_to_arize_phoenix_types(inputs) if inputs else {}
+        if processed_inputs:
+            child_span.set_attribute(SpanAttributes.INPUT_VALUE, self._safe_json_dumps(processed_inputs))
+            child_span.set_attribute(SpanAttributes.INPUT_MIME_TYPE, OpenInferenceMimeTypeValues.JSON.value)
+
+        processed_metadata = self._convert_to_arize_phoenix_types(metadata) if metadata else {}
+        if processed_metadata:
+            for key, value in processed_metadata.items():
+                child_span.set_attribute(f"{SpanAttributes.METADATA}.{key}", value)
+
+        self.child_spans[trace_id] = child_span
```

Comment:
```suggestion
        self.child_spans[UUID(trace_id)] = child_span
```
mypy requires a conversion to match its type definition.

Reply:
It's not required as I have defined the types for `child_spans` to be `[str, Span]` now.